go/vt/vttablet/onlineddl/executor.go

```diff
@@ -3784,8 +3787,10 @@ func (e *Executor) reviewRunningMigrations(ctx context.Context) (countRunnning i
 				if err := e.cutOverVReplMigration(ctx, s, shouldForceCutOver); err != nil {
 					_ = e.updateMigrationMessage(ctx, uuid, err.Error())
 					log.Errorf("cutOverVReplMigration failed: err=%v", err)
-					if merr, ok := err.(*sqlerror.SQLError); ok {
-						switch merr.Num {
+
+					if sqlErr, isSQLErr := sqlerror.NewSQLErrorFromError(err).(*sqlerror.SQLError); isSQLErr && sqlErr != nil {
```

Comment:
This change is required because the underlying SQL error is now likely to be wrapped by a vitess error.

Reply:
Not sure if we have any wrapping, but I think places where we cast errors directly probably should use https://pkg.go.dev/errors#example-As these days. So if we're changing this, maybe also switch to that?

Reply:
`sqlerror.NewSQLErrorFromError` is even a bit more elaborate and can parse a MySQL out of a string. We use that pretty consistently across the repo.
I'd say if we wanted to change that, it would be in a dedicated PR in one giant sweep.

Reply:
> sqlerror.NewSQLErrorFromError

It's not about this function though? It's about the cast after?

Reply:
Ah, the cast at that same line... Rich story. We tried to get rid of it altogether, but hit the (weird, IMO) `golang` design decision where an interface (`SQLError`) can be non-`nil` where the underlying implementation is `nil. So we kind of have to use the cast. What we know, though, is that the result of `sqlerror.NewSQLErrorFromError` is either a fully unwrapped `sqlerror.SQLError` or it is `nil`. So no need to use `As` on the result of this function.

Reference: https://github.com/vitessio/vitess/pull/12574